Sources/MongoSwift/Operations/NextOperation.swift

```diff
@@ -1,11 +1,20 @@
 import mongoc
 
-/// An operation corresponding to a `next` call on a MongoCursor.
+/// The entity on which the `next` operation is called.
+internal enum NextOperationTarget<T: Codable> {
+    /// Indicates the `next` call will be on a MongoCursor.
+    case cursor(MongoCursor<T>)
+
+    /// Indicates the `next` call will be on a change stream.
+    case changeStream(ChangeStream<T>)
+}
+
+/// An operation corresponding to a `next` call on a `NextOperationTarget`.
 internal struct NextOperation<T: Codable>: Operation {
-    private let cursor: MongoCursor<T>
+    private let target: NextOperationTarget<T>
 
-    internal init(cursor: MongoCursor<T>) {
-        self.cursor = cursor
+    internal init(target: NextOperationTarget<T>) {
+        self.target = target
     }
 
     internal func execute(using connection: Connection, session: ClientSession?) throws -> T? {
```

Comment:
I decided to combine `ChangeStreamNextOperation` and `NextOperation` because there was a lot of repeated code, but I am now running into a problem with the linter's Cyclomatic Complexity Violation. Is there another way to structure this code that is less complex than what I am doing?

Reply:
combining seems like a good idea to me. 

I'm not sure of a better way to structure this, I'd be fine with just turning off the cyclomatic complexity rule here. 

you might consider trying to identify all of the shared properties of these types and putting them on a protocol, something like:
```swift
protocol CursorLike {
    associatedtype OutputType: Decodable
    var state: CursorLikeState { get } // you'd have to start using same state type for both cursor and changestream
    static var closedError: Error { get }
    var nextFunc: (OpaquePointer, UnsafeMutablePointer<BSONPointer?>) -> (Bool) { get }
    var decoder: BSONDecoder { get }
}
```

(poorly named but you get the idea)

then you could make `NextOperation` generic on this type and you could do stuff like
```swift
guard case let .open(ptr, _, _, _) = target.state else {
    throw target.closedError
}

guard nextFunc(ptr, out) else {
    return nil
}

return target.decoder.decode(CursorType.OutputType.self, from: doc)
```

that said, it might not be worth it to do all that, but interesting to consider!

Reply:
Got it, I'll turn off the swiftlint rule in this case, but will keep this protocol method in mind for the future!